google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java

```diff
@@ -277,13 +286,40 @@ public static ApiTracerFactory createBigtableTracerFactory(
                     .build()))
         // Add OpenCensus Metrics
         .add(MetricsTracerFactory.create(tagger, stats, attributes))
-        .add(BuiltinMetricsTracerFactory.create(builtinAttributes))
         // Add user configured tracer
         .add(settings.getTracerFactory());
-
+    Attributes otelAttributes =
+        Attributes.of(PROJECT_ID, projectId, INSTANCE_ID, instanceId, APP_PROFILE, appProfileId);
+    BuiltinMetricsTracerFactory builtinMetricsTracerFactory =
+        setupBuiltinMetricsTracerFactory(settings.toBuilder(), otelAttributes);
+    if (builtinMetricsTracerFactory != null) {
+      tracerFactories.add(builtinMetricsTracerFactory);
+    }
     return new CompositeTracerFactory(tracerFactories.build());
   }
 
+  private static BuiltinMetricsTracerFactory setupBuiltinMetricsTracerFactory(
```

Comment:
This doesnt seem to have side effects other than just instantiating the BuiltinMetricsTracerFactory. Also I would pull out BuiltinMetricsTracerFactory creation out and keep this method focused on getting an instance of OpenTelemetry:

```java
createBigtableTracerFactory() {
 ...
  @Nullable OpenTelemetry otel = getOpenTelemetryInstance(settings);
   if (otel != null) {
      Attributes attrs =
        Attributes.of(PROJECT_ID, projectId, INSTANCE_ID, instanceId, APP_PROFILE, appProfileId);
      tracerFactories.add(BuiltinMetricsTracerFactory.create(otel, attrs);
   }
}

getOpenTelemetryInstance(EnhancedBigtableStubSettings. settings) {
  if (settings.getMetricsProvider instanceof CustomOpenTelemetryProvider) {
     return (CustomOpenTelemetryMetricsProvider) settings.getMetricsProvider();
  } else if ...
}
```

Reply:
refactor the code and rename the method